openedx/core/djangoapps/user_api/accounts/utils.py

```diff
@@ -200,3 +200,13 @@ def is_multiple_user_enterprises_feature_enabled():
         Boolean value representing switch status
     """
     return user_api_waffle().is_enabled(ENABLE_MULTIPLE_USER_ENTERPRISES_FEATURE)
+
+
+def is_multiple_sso_accounts_association_to_saml_user_enabled():
+    """
+    Checks to see if the django-waffle switch for enabling the multiple sso accounts association to saml user is active
```

Comment:
The purpose of this switch isn't really clear to me -- are we adding a control to determine if it is possible for a single user to be associated with more than one third party identity (ie, UserSocialAuth record)?  This is already the case -- lots of users have more than one USA record -- are we altering the default behavior?

Reply:
For the safe side, I have added this waffle switch. As, we need some time to test these changes on stage and meanwhile, changes will go on PROD. So, they have been kept behind a switch, which will be turned on once the changes have been tested on Stage.

Reply:
I guess I still don't see what this has to do with multiple SSO accounts.  We're simply checking to see if the learner is already linked to an enterprise customer, and if the IdP is also linked to that customer.

Reply:
This is just a flag which will be controled from admin panel and will be switched on on PROD once we are done with the testing of this change on stage. It doesn't have to do anything with functionality.

Reply:
🤷 -- will it be removed after the change has been validated?

Reply:
Yes, this can be removed once the changes are validated. Until then we can plug in and out the code by just clicking on a flag in the admin panel which is the safest way to take these changes.

Reply:
@mattdrayer Zaman has added this flag at my request, due to my concern for affecting third party auth in prod while we are testing the changes in stage. We'll remove the flag when we go to prod.